app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -14,17 +14,30 @@ def validate_config_info(options)
                 else
                   config_info[:transformation_mapping]
                 end
-
-      raise _('Must provide an existing transformation mapping') if mapping.blank?
+      # mapping.invalid? added as a part of validating migration plan
+      raise _('Must provide an existing transformation mapping') if mapping.blank? && mapping.invalid?
 
       pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
       post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
 
+      # Add as a part of validating migration plan
+      # check service_type. Filter type=ServiceTemplateAnsiblePlaybook
+      if pre_service_id != nil && !pre_service_id.empty?
+        pre_service_type  = ServiceTemplate.find(pre_service_id).type
+        raise _('Premigration service type MUST be \"ServiceTemplateAnsiblePlaybook\"') if pre_service_type != "ServiceTemplateAnsiblePlaybook"
+      end
+      if post_service_id != nil && !post_service_id.empty?
+        post_service_type = ServiceTemplate.find(post_service_id).type
+        raise _('Postmigration service type MUST be \"ServiceTemplateAnsiblePlaybook\"') if post_service_type != "ServiceTemplateAnsiblePlaybook"
+      end
+
       vms = []
       if config_info[:actions]
         vm_objects = VmOrTemplate.where(:id => config_info[:actions].collect { |vm_hash| vm_hash[:vm_id] }.compact).index_by(&:id).stringify_keys
         config_info[:actions].each do |vm_hash|
           vm_obj = vm_objects[vm_hash[:vm_id]] || vm_hash[:vm]
+          # vm_obj.invalid? added as a part of validating migration plan
+          raise _("Invalid VM found #{vm_obj.name}") if vm_obj.invalid?
```

Comment:
I think we should move this down a line, no point in checking validity if it's nil.

Reply:
done.